controlplane/src/core/plugins/metrics.ts

```diff
@@ -0,0 +1,52 @@
+import client, { Registry } from 'prom-client';
+import fp from 'fastify-plugin';
+import Fastify, { FastifyRequest, FastifyReply } from 'fastify';
+import { FastifyListenOptions } from 'fastify/types/instance.js';
+
+declare module 'fastify' {
+  interface FastifyInstance {
+    metricsServer: Fastify.FastifyInstance;
+    startMetricsServer(opts: FastifyListenOptions): Promise<string>;
+  }
+}
+
+export interface MetricsPluginOptions {
+  enabled?: boolean;
+  registry?: Registry;
+  path?: string;
+}
+
+export default fp<MetricsPluginOptions>(function (fastify, { path = '/metrics', registry = new Registry() }, next) {
+  const metricsLogger = fastify.log.child({
+    component: '@wundergraph/controlplane',
```

Comment:
I like the idea but we haven't done this somewhere else on the controlplane. I suggest to align it everywhere or leave it for now. Also logs are already scoped to a particular service. We wouldn't win any context by flagging logs that they come from the service.